### PR TITLE
pythonPackages.flask-httpauth: 4.4.0 -> 4.5.0

### PR DESCRIPTION
**Pull request by oxzi**
###### Motivation for this change
Bump Flask-HTTPAuth to its latest release.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 145054` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>onionshare</li>
    <li>onionshare-gui</li>
    <li>python38Packages.flask-httpauth</li>
    <li>python39Packages.flask-httpauth</li>
    <li>updog</li>
  </ul>
</details>

File: pkgs/development/python-modules/flask-httpauth/default.nix [hunk 1]
```diff
@@ -1,16 +1,18 @@
-{ lib, python, isPy3k, buildPythonPackage, fetchPypi, flask }:
+{ lib, python, isPy3k, buildPythonPackage, fetchPypi, pytest, flask }:
 
 buildPythonPackage rec {
   pname = "Flask-HTTPAuth";
```
Comment (fabaff) [inline pkgs/development/python-modules/flask-httpauth/default.nix:4]:
```suggestion
  pname = "flask-httpauth";
```

We should follow [PEP-503](https://nixos.org/manual/nixpkgs/stable/#contributing) here.

Comment (oxzi) [reply pkgs/development/python-modules/flask-httpauth/default.nix]: Thanks for your feedback. I changed the package name and the source's `pname` accordingly.

**Comment by oxzi** [comment]
Result of `nixpkgs-review pr 145054` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>onionshare</li>
    <li>onionshare-gui</li>
    <li>python38Packages.flask-httpauth</li>
    <li>python39Packages.flask-httpauth</li>
    <li>updog</li>
  </ul>
</details>

File: pkgs/development/python-modules/flask-httpauth/default.nix [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 
   src = fetchPypi {
-    inherit pname version;
-    sha256 = "0fl1if91hg2c92b6sic7h2vhxxvb06ri7wflmwp0pfiwbaisgamw";
+    pname = "Flask-HTTPAuth";
+    version = version;
+    sha256 = "0ada63rkcvwkakjyx4ay98fjzwx5h55br12ys40ghkc5lbyl0l1r";
   };
 
+  checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-httpauth/default.nix:15]: Please use pytestCheckHook,

File: pkgs/development/python-modules/flask-httpauth/default.nix [hunk 3]
```diff
@@ -1,16 +1,19 @@
-{ lib, python, isPy3k, buildPythonPackage, fetchPypi, flask }:
+{ lib, python, isPy3k, buildPythonPackage, fetchPypi, pytest, flask }:
 
 buildPythonPackage rec {
-  pname = "Flask-HTTPAuth";
-  version = "4.4.0";
+  pname = "flask-httpauth";
+  version = "4.5.0";
 
   disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-httpauth/default.nix:7]: Please use pythonOlder.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/flask-httpauth/default.nix (hunk 2 above)
Comment (oxzi) [reply pkgs/development/python-modules/flask-httpauth/default.nix]: Thanks, I have updated the PR.

**Comment by oxzi** [comment]
@SuperSandro2000: Would you please take a second look. Thanks.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 145054` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>onionshare</li>
<li>onionshare-gui</li>
<li>python38Packages.flask-httpauth</li>
<li>python39Packages.flask-httpauth</li>
<li>updog</li>
</ul>
</details>

**Pull request merged**